### PR TITLE
Hw 4

### PR DESCRIPTION
**Pull request by AntonBakaiev**

### CODE REVIEW

File: students/AntonBakaiev/hw-4/answer.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+let initialText = prompt('Введите урывок текста');
+
+
+function searchVowels(initialText) { 
+    let count = 0;
+    const vowels = ['a', 'e', 'i', 'o', 'u', 'y']; // проверочный массив гласных 
+    
+    for (let char of initialText.toLowerCase()) { // берем символ из строки и приводим к строчным буквам что бы не вносить в массив проверки доплнительно заглавные буквы
```
Comment (Temu4) [inline students/AntonBakaiev/hw-4/answer.js:8]: It's better to normalize initial string only one time than during every iteraction.
```suggestion
    const normalizedInitialText = initialText.toLowerCase();
    for (let char of normalizedInitialText) { // берем символ из строки и приводим к строчным буквам что бы не вносить в массив проверки доплнительно заглавные буквы
```

File: students/AntonBakaiev/hw-4/answer.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+const validCreditCard = /^[3][4|7]\d{13}$|^[5][1-5]\d{14}$|^[4]\d{15}$/;
+
+let phoneNumber = prompt("Enter your number phone");
+Validation(phoneNumber)
+let mail = prompt("Enter your E-mail ");
+Validation(mail)
+let creditCard = prompt("Enter your number credit card ");
+Validation(creditCard)
+
+function Validation(textForValidation) {
```
Comment (Temu4) [inline students/AntonBakaiev/hw-4/answer.js:36]: Please name function in `camelCase`. Also better to use verbs.
```suggestion
function validateInput(textForValidation) {
```

File: students/AntonBakaiev/hw-4/answer.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+//   телефон ^[\+]\d{3}\s\d{4}\s\d{2}\s\d{2}$
+//   мыло ^([a-z0-9_-]+\.)*[a-z0-9_-]+@[a-z0-9_-]+(\.[a-z0-9_-]+)*\.[a-z]{2,6}$
+//   кредитка  ^[3][4|7]\d{13}$|^[5][1-5]\d{14}$|^[4]\d{15}$
+
+const validMobilePhone = /^[\+]\d{3}\s\d{4}\s\d{2}\s\d{2}$/;   //valid format: +000 0000 00 00
+const validEMailAddress = /^([a-z0-9_-]+\.)*[a-z0-9_-]+@[a-z0-9_-]+(\.[a-z0-9_-]+)*\.[a-z]{2,6}$/;
+const validCreditCard = /^[3][4|7]\d{13}$|^[5][1-5]\d{14}$|^[4]\d{15}$/;
+
+let phoneNumber = prompt("Enter your number phone");
+Validation(phoneNumber)
```
Comment (Temu4) [inline students/AntonBakaiev/hw-4/answer.js:30]:
```suggestion
validateInput(phoneNumber)
```

File: students/AntonBakaiev/hw-4/answer.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+let mail = prompt("Enter your E-mail ");
+Validation(mail)
```
Comment (Temu4) [inline students/AntonBakaiev/hw-4/answer.js:32]:
```suggestion
validateInput(mail)
```

File: students/AntonBakaiev/hw-4/answer.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+let creditCard = prompt("Enter your number credit card ");
+Validation(creditCard)
```
Comment (Temu4) [inline students/AntonBakaiev/hw-4/answer.js:34]:
```suggestion
validateInput(creditCard)
```

File: students/AntonBakaiev/hw-4/answer.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+// регулярные выражения:
+//   телефон ^[\+]\d{3}\s\d{4}\s\d{2}\s\d{2}$
+//   мыло ^([a-z0-9_-]+\.)*[a-z0-9_-]+@[a-z0-9_-]+(\.[a-z0-9_-]+)*\.[a-z]{2,6}$
+//   кредитка  ^[3][4|7]\d{13}$|^[5][1-5]\d{14}$|^[4]\d{15}$
+
+const validMobilePhone = /^[\+]\d{3}\s\d{4}\s\d{2}\s\d{2}$/;   //valid format: +000 0000 00 00
+const validEMailAddress = /^([a-z0-9_-]+\.)*[a-z0-9_-]+@[a-z0-9_-]+(\.[a-z0-9_-]+)*\.[a-z]{2,6}$/;
+const validCreditCard = /^[3][4|7]\d{13}$|^[5][1-5]\d{14}$|^[4]\d{15}$/;
+
+let phoneNumber = prompt("Enter your number phone");
```
Comment (Temu4) [inline students/AntonBakaiev/hw-4/answer.js:29]: Why `let`?

```suggestion
const phoneNumber = prompt("Enter your number phone");
```

**Review by Temu4 — CHANGES_REQUESTED** [review]

File: students/AntonBakaiev/hw-4/answer.js (hunk 1 above)
Comment (AntonBakaiev) [reply students/AntonBakaiev/hw-4/answer.js]: Виконав правки

File: students/AntonBakaiev/hw-4/answer.js (hunk 3 above)
Comment (AntonBakaiev) [reply students/AntonBakaiev/hw-4/answer.js]: +

File: students/AntonBakaiev/hw-4/answer.js (hunk 4 above)
Comment (AntonBakaiev) [reply students/AntonBakaiev/hw-4/answer.js]: +

File: students/AntonBakaiev/hw-4/answer.js (hunk 5 above)
Comment (AntonBakaiev) [reply students/AntonBakaiev/hw-4/answer.js]: +

File: students/AntonBakaiev/hw-4/answer.js (hunk 6 above)
Comment (AntonBakaiev) [reply students/AntonBakaiev/hw-4/answer.js]: Зробив правки! Не подумавши зробив let краще буде const

File: students/AntonBakaiev/hw-4/answer.js (hunk 2 above)
Comment (AntonBakaiev) [reply students/AntonBakaiev/hw-4/answer.js]: +

**Pull request merged**